### ISSUE TITLE
Parameter name: length,  Length cannot be less than zero error

### ISSUE THREAD

**Issue by lowrylp** [GZidar/CorePDF#19]
The following code is generating the Length cannot be less that zero error.  This basically from your example.  It occurs on the publish document.  Any idea why?

        public static string GenerateReceipt(string recNumber, string webRoot)
        {
            try
            {
                Document doc = new Document();

                doc.Pages.Add(new Page
                {
                    PageSize = Paper.PAGEA4PORTRAIT,
                    Contents = new List<Content>()
                    {
                        new TextBox
                        {
                            Text = "Receipt Number is:" + recNumber,
                            FontFace = "Helvetica",
                            FontSize = 14,
                            PosX = 750,
                            PosY = 100,
                            TextAlignment = Alignment.Left
                        },
                        new Shape
                        {
                            Type = Polygon.Rectangle,
                            PosX = 0,
                            PosY = 0,
                            Height = 842,
                            Width = 595,
                            FillColor = "#ffffff"
                        }
                    }
                });

                var file = System.IO.Path.Combine(webRoot, recNumber + ".pdf");

                using (var filestream = new FileStream(file, FileMode.Create, FileAccess.Write))
                {
                    doc.Publish(filestream);
                }

                return recNumber + ".pdf";
            }
            catch (System.Exception ex)
            {
                Console.WriteLine("Receipt " + ex.Message);
                return string.Empty;
            }
        }

**Comment by lowrylp** [comment]
Forget to say I am using version 0.4.0-beta from NuGet.

**Comment by GZidar** [comment]
Hi @lowrylp, thanks for letting me know. I'll take a look and get back to you.

**Comment by GZidar** [comment]
I was able to reproduce this and it's because of the value you have set for PosX exceeds the width of the page. A4 Portrait has Width of 595 and Height of 842, your example code has set the PosX at 750.


If you want the Receipt number to at the extreme right, you could set the PosX to 580 and Set TextAlignment to be Right. That avoids the error.


I'll add some more useful error message in the future so that issues like this are clearer. 


I hope that helps.

**Comment by lowrylp** [comment]
Thank you sir for pointing out my error.  I just got my X's and Y's messed up.
A more useful error message would be helpful.

Might I ask another question?
The  pdf is created through a stream to the output file.   I have tried pushing that
stream to  another stream so that the pdf generated can be sent to Amazon S3.
But I cannot get it to work.  It seems like the pdf stream is terminated and thus
nothing is sent to S3.    Have you tried anything like this before?

Thanks again.

Larry
